lms/djangoapps/bookmarks/tests/test_views.py

```diff
@@ -148,54 +168,84 @@ class BookmarksListViewTests(BookmarksViewTestsMixin):
     POST /api/bookmarks/v0/bookmarks
     """
     @ddt.data(
-        ('course_id={}', False),
-        ('course_id={}&fields=path,display_name', True),
+        (1, False),
+        (10, False),
+        (100, False),
+        (1, True),
+        (10, True),
+        (100, True),
     )
     @ddt.unpack
-    def test_get_bookmarks_successfully(self, query_params, check_optionals):
+    def test_get_bookmarks_successfully(self, bookmarks_count, check_all_fields):
         """
         Test that requesting bookmarks for a course returns records successfully in
         expected order without optional fields.
         """
-        response = self.send_get(
-            client=self.client,
-            url=reverse('bookmarks'),
-            query_parameters=query_params.format(urllib.quote(self.course_id))
-        )
+
+        course, blocks, bookmarks_data = self.create_course_with_bookmarks_count(bookmarks_count)
+
+        query_parameters = 'course_id={}&page_size={}'.format(urllib.quote(unicode(course.id)), 100)
+        if check_all_fields:
+            query_parameters += '&fields=path,display_name'
+
+        with self.assertNumQueries(7):
```

Comment:
It's often helpful to list out the 7 expected queries, so that the next time someone comes to this line, they know what queries to look for.